presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -721,7 +721,7 @@ public static long levenshteinDistance(@SqlType("varchar(x)") Slice left, @SqlTy
     @SqlType(StandardTypes.BIGINT)
     public static long hammingDistance(@SqlType("varchar(x)") Slice left, @SqlType("varchar(y)") Slice right)
     {
-        int distance = 0;
+        long distance = 0;
```

Comment:
If we think asymptotically, it will require a bunch of time to do a linear scan until LONG.MAX_VALUE(~10^18). I won't even mention how much space such varchar string can take. I think the return type should be `Int.MaxValue` as Long type is not feasible here.